pennylane/qnode.py

```diff
@@ -728,6 +728,9 @@ def tape(self) -> QuantumTape:
     def construct(self, args, kwargs):
         """Call the quantum function with a tape context, ensuring the operations get queued."""
 
+        if self.interface == "auto":
+            self.interface = qml.math.get_interface(*args, *list(kwargs.values()))

```

Comment:
Need to set interface back to "auto"

Reply:
I don't think so, we cannot pass it to execute if we reset in the construct and therefore we would need to recheck the interface in `execute` (currently it is reset at the end of the QNode call). Do you have a use case in mind?

Reply:
When calling a `QNode`, the `__call__` method switches from "auto" to the correct interface, and then switches back to "auto". But what if a user calls `construct` directly? Then we would lose the "auto"interface.

Reply:
Thanks done :+1: